pandas/core/indexes/multi.py

```diff
@@ -3049,8 +3050,69 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        indexer = self._reorder_indexer(seq, indexer)
+
         return indexer._ndarray_values
 
+    def _reorder_indexer(
+        self, seq: Tuple[Union[Scalar, Iterable, AnyArrayLike], ...], indexer: ArrayLike
+    ) -> ArrayLike:
+        """
+        Reorder an indexer of a MultiIndex (self) so that the label are in the
+        same order as given in seq
+
+        Parameters
+        ----------
+        seq : label/slice/list/mask or a sequence of such
+        indexer: an Int64Index indexer of self
+
+        Returns
+        -------
+        indexer : a sorted Int64Index indexer of self ordered as seq
+        """
+        # Find out if the list_like label are sorted as the levels or not
+        need_sort = False
+        for i, k in enumerate(seq):
+            if is_list_like(k):
+                if not need_sort:
+                    k_codes = self.levels[i].get_indexer(k)
+                    k_codes = k_codes[k_codes >= 0]  # Filter absent keys
+                    # True if the given codes are not ordered
+                    need_sort = (k_codes[:-1] > k_codes[1:]).any()
+        # Bail out if no need to sort
+        # This is only true for a lexsorted index
```

Comment:
can't we just test is_lexsorted?

Reply:
isn't that sufficient here?

Reply:
The property we need to test is "do the index and the selection share the same order".
If we have 
```
df = pd.DataFrame(np.arange(2), index=[["b", "a"]]) 
df
   0
b  0
a  1

```
Then the index is not lexsorted, and we do not need to sort when doing
`df.loc[['b', 'a']]`, but we need to sort for getting `df.loc[['a', 'b']]` in the right order.

But they here room for improvement in this place, and I will propose something.

Reply:
So I made a minor modification, but is the index is lexsorted, we still need to check if the requested order is also sorted before bailing out.

A another possibility is to sort in every case, if the performance hit is acceptable.